sky/clouds/oci.py

```diff
@@ -501,59 +511,45 @@ def _get_image_id(
         region_name: str,
         instance_type: str,
     ) -> str:
-        if image_id is None:
-            return self._get_default_image(region_name=region_name,
-                                           instance_type=instance_type)
-        if None in image_id:
-            image_id_str = image_id[None]
-        else:
-            assert region_name in image_id, image_id
-            image_id_str = image_id[region_name]
-        if image_id_str.startswith('skypilot:'):
+        image_id_str = self._get_image_tag(image_id=image_id,
+                                           instance_type=instance_type,
+                                           region=region_name)
+
+        if not image_id_str.startswith('ocid1.image'):
             image_id_str = service_catalog.get_image_id_from_tag(image_id_str,
                                                                  region_name,
                                                                  clouds='oci')
-            if image_id_str is None:
-                logger.critical(
-                    '! Real image_id not found! - {region_name}:{image_id}')
-                # Raise ResourcesUnavailableError to make sure the failover
-                # in CloudVMRayBackend will be correctly triggered.
-                # TODO(zhwu): This is a information leakage to the cloud
-                # implementor, we need to find a better way to handle this.
-                raise exceptions.ResourcesUnavailableError(
-                    '! ERR: No image found in catalog for region '
-                    f'{region_name}. Try setting a valid image_id.')
+
+        # Image_id should be impossible be None, except for the case when
+        # user specify an image tag which does not exist in the image.csv
+        # catalog file which only possible in "test" / "evaluation" phase.
+        # Therefore, we use assert here.
+        assert image_id_str is not None
 
         logger.debug(f'Got real image_id {image_id_str}')
         return image_id_str
 
-    def _get_default_image(self, region_name: str, instance_type: str) -> str:
+    def _get_image_tag(self, image_id: Optional[Dict[Optional[str], str]],
```

Comment:
`tag` is a special term we used for images with `skypilot:` prefix. Let's avoid confusing that : )
```suggestion
    def _get_image_str(self, image_id: Optional[Dict[Optional[str], str]],
```